examples/custom_agent/custom_agent.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright 2021-present Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <uxr/agent/transport/custom/CustomAgent.hpp>
+#include <uxr/agent/transport/endpoint/IPv4EndPoint.hpp>
+
+#include <poll.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <signal.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+/**
+ * This custom XRCE Agent example attempts to show how easy is for the user to define a custom
+ * Micro XRCE-DDS Agent behaviour, in terms of transport initialization and closing, and also
+ * regarding read and write operations.
+ * For this simple case, an UDP socket is opened on port 8888. Additionally, some information
+ * messages are being printed to demonstrate the custom behaviour.
+ * As the endpoint is already defined, we are using the provided
+ * `eprosima::uxr::IPv4EndPoint` by the library.
+ * Other transport protocols might need to implement their own endpoint struct.
+ */
+
+int main(int argc, char** argv)
+{
+    eprosima::uxr::Middleware::Kind mw_kind(eprosima::uxr::Middleware::Kind::FASTDDS);
+    uint16_t agent_port(8888);
+    struct pollfd poll_fd;
+
+    /**
+     * @brief Agent's initialization behaviour description.
+     */
+    eprosima::uxr::CustomAgent::InitFunction init_function = [&]() -> bool
```

Comment:
Is not possible to pass some argument to the init function in order to specify transport-related parameters?

Reply:
The CustomEndPoint is just meant to be used for identifying messages during receive/send operations. Therefore I do not think there is any simpler way of doing this than just capture by reference required variables from the main loop that the user needs to declare to init/fini the transport, as it is done in the example with `poll_fd` or `agent_port`